artiq/applets/plot_xy.py

```diff
@@ -24,32 +31,57 @@ def data_changed(self, data, mods, title):
         fit = data.get(self.args.fit, (False, None))[1]
 
         if not len(y) or len(y) != len(x):
-            return
+            self.mismatch['X values'] = True
+            self.timer.start(1000)
```

Comment:
This restarts the timer, so regular data input that constantly produces mismatches will not trigger the warning.
I think your previous PR also has this bug.

Reply:
> flag -> timer_active (with the opposite True/False values)?
Though:
> This could be replaced with timer.isActive()
Is it necessary to have this check at all?

In the last PR, I remove the flag checking. But the purpose of it is to make sure that the warning is shown right after the timeout of the happening of the first failure.

Reply:
With your previous PR, if you regularly send data that produces a mismatch every time, it will not trigger the warning AFAICT (but it should). This is due to the timer being constantly restarted before it can fire.